lib/extension/window.js

```diff
@@ -68,49 +68,67 @@ export class WindowManager extends GObject.Object {
       !production ? "DEV" : `${PACKAGE_VERSION}-${ext.metadata.version}`
     }`;
     this.windowProps = this.ext.configMgr.windowProps;
+    this.windowProps.overrides = this.windowProps.overrides.filter(
+      (override) => (!override.wmId)
+    );
     this._kbd = this.ext.keybindings;
     this._tree = new Tree(this);
     this.eventQueue = new Queue();
     this.theme = this.ext.theme;
     Logger.info("forge initialized");
   }
 
-  addFloatOverride(metaWindow, byClass = true) {
+  addFloatOverride(metaWindow, byClass, nonPersistent) {
     let overrides = this.windowProps.overrides;
     let wmTitle = metaWindow.get_title();
     let wmClass = metaWindow.get_wm_class();
+    let wmId = metaWindow.get_id();
 
     for (let override in overrides) {
-      if (!byClass) {
-        if (override.wmClass === wmClass && override.wmTitle === wmTitle) return;
-      } else {
-        if (override.wmClass === wmClass && !override.wmTitle) return;
-      }
+      // if the window is already floating
+      if (override.wmClass === wmClass &&
+        override.mode === "float" &&
+        (
+          (byClass && !override.wmTitle) ||
+          (!byClass && override.wmTitle === wmTitle)
+        )
+      ) return;
+    }
+    if (nonPersistent) {
+      overrides.push({
+        wmClass: wmClass,
+        wmTitle: !byClass ? wmTitle : undefined,
+        wmId: wmId,
+        mode: "float"
+      });
+    } else {
+      overrides.push({
+        wmClass: wmClass,
+        wmTitle: !byClass ? wmTitle : undefined,
+        mode: "float",
+      });
```

Comment:
It may be better :

```suggestion
    overrides.push({
      wmClass: wmClass,
      wmTitle: !byClass ? wmTitle : undefined,
      wmId: nonPersistent? wmId: undefined,
      mode: "float"
    });
```